src/tests/test_consumer.py

```diff
@@ -0,0 +1,571 @@
+        ],
+        test_commands=[],
+    ),
+    ConsumerRepo(
+        name="module_template",
+        git_url="git@github.com:eclipse-score/module_template.git",
+        commands=[
+            "bazel run //docs:ide_support",
+            "bazel run //docs:incremental",
+            "bazel build //docs:docs",
+        ],
+        test_commands=[
+            "bazel test //tests/...",
+        ],
+    ),
+]
+
+
+@pytest.fixture(scope="session")
+def sphinx_base_dir(tmp_path_factory: TempPathFactory) -> Path:
+    return tmp_path_factory.mktemp("testing_dir")
+
+
+def get_current_git_commit(curr_path: Path):
+    """
+    Get the current git commit hash (HEAD).
+    """
+    result = subprocess.run(
+        ["git", "rev-parse", "HEAD"],
+        capture_output=True,
+        text=True,
+        check=True,
+        cwd=curr_path,
+    )
+    return result.stdout.strip()
+
+
+def get_remote_main_commit(curr_path: Path) -> str:
+    """
+    Get the commit hash of the remote main branch, to find possible metamodel changes
+    """
+    _ = subprocess.run(
+        ["git", "fetch", "origin", "main"],
+        capture_output=True,
+        text=True,
+        check=True,
+        cwd=curr_path,
+    )
+
+    result = subprocess.run(
+        ["git", "rev-parse", "origin/main"],
+        capture_output=True,
+        text=True,
+        check=True,
+        cwd=curr_path,
+    )
+    LOGGER.debug(f"Grabbed remote origin/main with commit: {result.stdout.strip()}")
+    return result.stdout.strip()
+
+
+def check_for_metamodel_change(curr_path: Path, remote_base_commit: str):
+    """
+    Check if there are any changes in the metamodel.yaml file.
+    """
+    try:
+        # Get list of changed files
+        result = subprocess.run(
+            ["git", "diff", "--name-only", remote_base_commit, "HEAD"],
+            capture_output=True,
+            text=True,
+            check=True,
+            cwd=curr_path,
+        )
+
+        changed_files = result.stdout.strip().split("\n")
+
+        # Check if any changed files are in metamodel paths
+        for file_path in changed_files:
+            if "metamodel.yaml" in file_path:
+                LOGGER.debug(
+                    "Metamodel.yaml has changed. Ignoring 'score_metamodel' errors."
+                )
+                return True
+
+        LOGGER.debug("Metamodel.yaml has NOT changed. Parsing all errors")
+        return False
+
+    except Exception as e:
+        # Wondering if this is a good idea.
+        # Or better to fail here?
+        LOGGER.debug(f"Failed git diff. Error: {e}")
+        return False
+
+
+def replace_bazel_dep_with_local_override(module_content: str) -> str:
+    """ """
+
+    # Pattern to match the bazel_dep line
+    pattern = rf'bazel_dep\(name = "score_docs_as_code", version = "[^"]+"\)'
+
+    # Replacement with local_path_override
+    replacement = f"""bazel_dep(name = "score_docs_as_code", version = "0.4.2")
```

Comment:
Force failure without override:
```suggestion
    replacement = f"""bazel_dep(name = "score_docs_as_code", version = "0.0.0")
```